src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -328,6 +328,9 @@ class HMM
   template<typename Archive>
   void serialize(Archive& ar, const unsigned int version);
 
+  bool ApproximatelyEqual(const HMM<Distribution>& other, 
+                          double tolerance) const;
+
```

Comment:
Since this is used only for testing, could we instead add this in the test file only instead of the `HMM` code?

Reply:
I also hesitated while adding this function -- it didnt seem very clean. However, the HMMModel class doesn't expose pointers to HMM<Distribution>. Should I provide accessors like
``` c++
HMM<distribution::DiscreteDistribution> * DiscreteHMM() { return discreteHMM; }
HMM<distribution::GaussianDistribution> * GaussianHMM() { return gaussianHMM; }
HMM<gmm::GMM> * GMMHMM() { return gmmHMM; }
```
in the HMMModel class and then an ApproximatelyEqual() function in the test file? Is there another (better) way?

Reply:
Yeah, I think that is a reasonable way to do it, since others may have use for those four functions you are proposing.  I would ask that you document those functions and point out that the pointers for the other types of HMMs will be NULL... i.e. if the HMMModel holds a discrete HMM, then GaussianHMM() will return NULL and so will GMMHMM(), and the user should be careful.